File: scratch/search_transcript.py
import json

path = r"C:\Users\prach\.gemini\antigravity\brain\94682b3f-bee1-4a4f-97fd-35dded607fbd\.system_generated\logs\transcript.jsonl"
with open(path, "r", encoding="utf-8") as f:
    for line in f:
        data = json.loads(line)
        if data.get("source") == "USER_EXPLICIT" and "type" in data and data["type"] == "USER_INPUT":
            content = data.get("content", "")
            if "profit" in content.lower() or "report" in content.lower() or "loss" in content.lower():
                print(f"Index: {data.get('step_index')} | Date: {data.get('created_at')}")
                print(content)
                print("-" * 50)

File: scratch/search_transcript.php
<?php

$path = '/var/www/html/scratch/transcript.jsonl';
if (!file_exists($path)) {
    echo "Transcript file not found at $path\n";
    exit(1);
}

$handle = fopen($path, "r");
if ($handle) {
    while (($line = fgets($handle)) !== false) {
        $data = json_decode($line, true);
        if (isset($data['source']) && $data['source'] === 'USER_EXPLICIT' && isset($data['type']) && $data['type'] === 'USER_INPUT') {
            $content = $data['content'] ?? '';
            if (stripos($content, 'profit') !== false || stripos($content, 'report') !== false || stripos($content, 'loss') !== false || stripos($content, 'show') !== false) {
                echo "Index: " . ($data['step_index'] ?? '') . " | Date: " . ($data['created_at'] ?? '') . "\n";
                echo $content . "\n";
                echo str_repeat("-", 50) . "\n";
            }
        }
    }
    fclose($handle);
}
